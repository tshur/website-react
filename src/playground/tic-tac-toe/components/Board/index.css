:local .row > * {
  height: 60px;
  display: inline-block;
}