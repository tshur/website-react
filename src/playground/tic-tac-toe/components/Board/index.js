import React, { Component } from 'react';
import styles from './index.css';

import Square from '../Square';

class Board extends Component {
  constructor(props) {
    super(props);

    this.state = {
      squares: Array(9).fill(null)
    };

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(i) {
    const squares = this.state.squares.slice();
    squares[i] = 'X';
    this.setState({ squares });
  }

  renderSquare(i) {
    const { squares } = this.state;

    return (
      <Square
        value={squares[i]}
        onClick={() => this.handleClick(i)}
      />
    );
  }

  render() {
    return (
      <div>
        <div className={styles.row}>
          {this.renderSquare(0)}
          {this.renderSquare(1)}
          {this.renderSquare(2)}
        </div>
        <div className={styles.row}>
          {this.renderSquare(3)}
          {this.renderSquare(4)}
          {this.renderSquare(5)}
        </div>
        <div className={styles.row}>
          {this.renderSquare(6)}
          {this.renderSquare(7)}
          {this.renderSquare(8)}
        </div>
      </div>
    );
  }
}

export default Board;