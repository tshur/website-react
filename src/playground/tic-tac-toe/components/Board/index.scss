.board {
  display: table;
  border-collapse: collapse;

  .row {
    display: table-row;
    border-bottom: 3px solid #444;

    &:last-child {
      border-bottom: 0;
    }
  }
}


