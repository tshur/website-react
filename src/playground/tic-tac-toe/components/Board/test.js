import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Board from '.';

describe('Board', () => {
  const squares = Array(9).fill(null);

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Board
        squares={squares}
        onClick={() => ''}
      />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Board
        squares={squares}
        onClick={() => ''}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
