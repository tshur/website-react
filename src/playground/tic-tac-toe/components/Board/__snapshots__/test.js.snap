// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board has a valid snapshot 1`] = `
<div>
  <div
    key="0"
  >
    <Square
      key="0"
      onClick={[Function]}
      value={null}
      winningSquare={null}
    />
    <Square
      key="1"
      onClick={[Function]}
      value={null}
      winningSquare={null}
    />
    <Square
      key="2"
      onClick={[Function]}
      value={null}
      winningSquare={null}
    />
  </div>
  <div
    key="1"
  >
    <Square
      key="3"
      onClick={[Function]}
      value={null}
      winningSquare={null}
    />
    <Square
      key="4"
      onClick={[Function]}
      value={null}
      winningSquare={null}
    />
    <Square
      key="5"
      onClick={[Function]}
      value={null}
      winningSquare={null}
    />
  </div>
  <div
    key="2"
  >
    <Square
      key="6"
      onClick={[Function]}
      value={null}
      winningSquare={null}
    />
    <Square
      key="7"
      onClick={[Function]}
      value={null}
      winningSquare={null}
    />
    <Square
      key="8"
      onClick={[Function]}
      value={null}
      winningSquare={null}
    />
  </div>
</div>
`;
